packages/manager/apps/pci-load-balancer/src/api/hook/useGateways.ts

```diff
@@ -20,3 +29,61 @@ export const useGetSubnetGateways = (
     enabled: !!projectId && !!regionName && !!subnetId,
     throwOnError: true,
   });
+
+export const useRegionPlans = (regionName: string) => {
+  const { data: project } = useProject();
+  const { data } = useProductAvailability(project.project_id, {
+    addonFamily: 'gateway',
+  });
+
+  return useMemo(() => {
+    const plans = data?.plans.filter(({ code }) => code.includes('hour')) || [];
+    const regionPlans =
+      plans.filter((plan) =>
+        plan.regions.some((region) => region.name === regionName),
+      ) || [];
+
+    return regionPlans.map((item) => item.code);
+  }, [data, regionName]);
+};
```

Comment:
You could use functional programing to prevent from using intermediates constants, by piping functions: 

```suggestion
  return useMemo(
    () =>
      data?.plans.filter(({ code }) => code.includes('hour')) ||
      [].filter((plan) =>
        plan.regions.some((region) => region.name === regionName),
      ) ||
      [].map((item) => item.code),
    [data, regionName],
  );
};
```

Reply:
```suggestion
  return useMemo(
    () =>
      (
        (
          data?.plans.filter(({ code }) => code.includes('hour')) || []
        ).filter((plan) =>
          plan.regions.some((region) => region.name === regionName),
        ) || []
      ).map((item) => item.code),
    [data, regionName],
  )
};
```
I have to add parentheses to do it this way but I don't think it's easier to read than using intermediates constants no ?

Reply:
It's up to you :) but map(), filter(), etc.. are functions that have been implemented in a functional way, to remove unnecessary boilerplate code like constants that hold some memory usage to do nothing ;)

Reply:
BTW, I can't figure out why you have to add extra parenthesis, in my IDE it works well without

Reply:
It was not the IDE. If I does not put the parenthesis It does not work because `filter` will be on the empty array. But finally that does not matter, I found the simple way to improve it if it's good for you

`return useMemo(
    () =>
      data?.plans
        .filter(({ code }) => code.includes('hour'))
        .filter((plan) =>
          plan.regions.some((region) => region.name === regionName),
        )
        .map((item) => item.code) || [],
    [data, regionName],
  );`